src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -855,6 +872,76 @@ func validateListener(
 				external.Subdomain,
 				"TLS requires specifying a subdomain"))
 	}
+	if tlsEnabled && clientCACertRef != nil {
+		if !requireClientAuth {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("requireClientAuth"),
+					requireClientAuth,
+					"Enabled has to be set to true for RequireClientAuth if ClientCACertRef is set"))
+		}
+
+		if clientCACertRef.Name == "" {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("clientCACertRef"),
+					clientCACertRef,
+					"Name must be provided if ClientCACertRef is set"))
+		}
+
+		if clientCACertRef.Kind != "" && !strings.EqualFold(clientCACertRef.Kind, "Secret") {
+			allErrs = append(allErrs,
+				field.Invalid(
+					path.Child("clientCACertRef"),
+					clientCACertRef,
+					"Kind must be set to secret if set in ClientCACertRef"))
+		}
+
+		if len(allErrs) == 0 {
+			if clientCACertRef.Namespace == "" {
+				clientCACertRef.Namespace = clusterNamespace
+			}
+
+			secret := &corev1.Secret{}
+			err := kclient.Get(context.TODO(), types.NamespacedName{Name: clientCACertRef.Name, Namespace: clientCACertRef.Namespace}, secret)
```

Comment:
```suggestion
			err := kclient.Get(context.Background(), types.NamespacedName{Name: clientCACertRef.Name, Namespace: clientCACertRef.Namespace}, secret)
```

Reply:
I think we should context.TODO() per https://pkg.go.dev/context#Background.


```
Background returns a non-nil, empty Context. It is never canceled, has no values, and has no deadline. It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.


TODO returns a non-nil, empty Context. Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).
```